executor/metric_reader.go

```diff
@@ -193,3 +194,53 @@ func (c *queryClient) URL(ep string, args map[string]string) *url.URL {
 	ep = strings.Replace(ep, "api/v1", "pd/api/v1/metric", 1)
 	return c.Client.URL(ep, args)
 }
+
+// MetricSummaryRetriever uses to read metric data.
+type MetricSummaryRetriever struct {
+	dummyCloser
+	table     *model.TableInfo
+	extractor *plannercore.MetricTableExtractor
+	quantiles []float64
+	retrieved bool
+}
+
+func (e *MetricSummaryRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	if e.retrieved || e.extractor.SkipRequest {
+		return nil, nil
+	}
+	e.retrieved = true
+	totalRows := make([][]types.Datum, 0, len(infoschema.MetricTableMap))
+	e.quantiles = []float64{0.999, 0.99, 0.90, 0.80}
+	tps := make([]*types.FieldType, 0, len(e.table.Columns))
+	for _, col := range e.table.Columns {
+		tps = append(tps, &col.FieldType)
+	}
+	startTime := e.extractor.StartTime.Format(plannercore.MetricTableTimeFormat)
+	endTime := e.extractor.EndTime.Format(plannercore.MetricTableTimeFormat)
+	for name, def := range infoschema.MetricTableMap {
+		sqls := e.genMetricQuerySQLS(name, startTime, endTime, def.Quantile)
+		for _, sql := range sqls {
+			rows, _, err := sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			for _, row := range rows {
+				totalRows = append(totalRows, row.GetDatumRow(tps))
+			}
+		}
+	}
+	return totalRows, nil
+}
+
+func (e *MetricSummaryRetriever) genMetricQuerySQLS(name, startTime, endTime string, quantile float64) []string {
+	if quantile == 0 {
+		sql := fmt.Sprintf(`select "%s",min(time),sum(value),avg(value),min(value),max(value) from metric_schema.%s where time > '%s' and time < '%s'`, name, name, startTime, endTime)
```

Comment:
Does the sum value make sense?

Reply:
sometime it is useful, Such as `AWR` has `SUM CPU TIME`